cmd/containerid/main.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: (c) Copyright 2023 Advanced Micro Devices, Inc.
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/Xilinx-CNS/kubernetes-onload/pkg/containerid"
+)
+
+func getContainerID(
+	podNamespace string, podName string, containerName string,
+	apiTimeout time.Duration, apiPollingInternal time.Duration,
+) (string, error) {
+	for start := time.Now(); time.Since(start) < apiTimeout; {
```

Comment:
Just found what looks like a nicer way to handle timeouts with the standard library https://pkg.go.dev/context#WithTimeout.

Reply:
Actually, it is even simpler to handle the timeouts with the `timeout` tool from Shell.